src/lib/source-handlers/azure/list-repos.ts

```diff
@@ -0,0 +1,84 @@
+import { AzureRepoData } from './types';
+import Bottleneck from 'bottleneck';
+import base64 = require('base-64');
+import * as debugLib from 'debug';
+import { getAzureToken } from './get-azure-token';
+import { getAzureOrgName } from './get-azure-orgname';
+import * as needle from 'needle';
+
+const debug = debugLib('snyk:azure');
+const azureDefaultUrl = 'https://dev.azure.com';
+
+const limiter = new Bottleneck({
+  maxConcurrent: 1,
+  minTime: 500,
+});
+
+limiter.on('failed', async (error, jobInfo) => {
+  const id = jobInfo.options.id;
+  console.warn(`Job ${id} failed: ${error}`);
+  if (jobInfo.retryCount === 0) {
+    // Here we only retry once
+    console.log(`Retrying job ${id} in 25ms!`);
+    return 25;
+  }
+});
+
+async function fetchAllRepos(
+  url: string,
+  orgName: string,
+  project: string,
+  token: string,
+): Promise<AzureRepoData[]> {
+  debug('Fetching repos for ' + project);
+  const repoList: AzureRepoData[] = [];
+  const data = await limiter.schedule(() =>
+    needle(
+      'get',
+      `${url}/${orgName}/` +
+        encodeURIComponent(project) +
+        '/_apis/git/repositories?api-version=4.1',
+      {
+        headers: { Authorization: 'Basic ' + base64.encode(':' + token) },
```

Comment:
You can use string literals here, it is the new way to combine strings instead of `+`
`Basic ${base64.encode(':' + token)}`

Reply:
ok